project/src/test/java/caselab/service/users/ApplicationUserServiceTest.java

```diff
@@ -122,71 +136,91 @@ void updateUser_shouldUpdateAndReturnUserResponse() {
 
-        when(userRepository.findById(user1Id)).thenReturn(Optional.of(existingUser));
+        when(userRepository.findByEmail(existingUser.getEmail())).thenReturn(Optional.of(existingUser));
         when(authService.encodePassword(newPassword)).thenReturn("hashedPassword");
         when(userRepository.save(existingUser)).thenReturn(existingUser);
         when(userMapper.entityToResponse(existingUser)).thenReturn(updatedUserResponse);
 
-        UserResponse result = userService.updateUser(any(Authentication.class), updateRequest);
+        UserResponse result = userService.updateUser(authentication, updateRequest);
 
         assertEquals(updatedUserResponse, result);
         verify(authService, times(1)).encodePassword(newPassword);
         verify(userRepository, times(1)).save(existingUser);
     }
 
     @Test
-    void updateUser_shouldNotUpdatePasswordWhenEmpty() {
-        UserUpdateRequest updateRequestEmptyPassword = new UserUpdateRequest("john_updated", "");
+    void updateUser_shouldUpdatePassword() {
+        String newPassword = "NewPassword";
+        UserUpdateRequest updateRequest = createUserUpdateRequest("JohnNewName", newPassword);
 
-        UserResponse updatedUserResponse = UserResponse.builder().email("john_updated").build();
+        Authentication authentication = mock(Authentication.class);
+        UserDetails userDetails = mock(UserDetails.class);
 
-        when(userRepository.findById(user1Id)).thenReturn(Optional.of(user1));
-        when(userRepository.save(user1)).thenReturn(user1);
-        when(userMapper.entityToResponse(user1)).thenReturn(updatedUserResponse);
+        when(authentication.getPrincipal()).thenReturn(userDetails);
+        when(userDetails.getUsername()).thenReturn(userEmail);
 
-        UserResponse resultEmptyPassword = userService.updateUser(any(Authentication.class), updateRequestEmptyPassword);
+        ApplicationUser existingUser = createUser(user1Id, "johnDoe@gmail.com", "john_doe", "oldPassword");
+        UserResponse updatedUserResponse = createUserResponse("johnDoe@gmail.com", "john_updated", new ArrayList<>());
 
-        assertEquals(updatedUserResponse, resultEmptyPassword);
-        verify(authService, times(0)).encodePassword(any());
-        verify(userRepository, times(1)).save(user1);
-    }
+        when(userRepository.findByEmail(userEmail)).thenReturn(Optional.of(existingUser));
+        when(authService.encodePassword(newPassword)).thenReturn("hashedPassword");
+        when(userRepository.save(existingUser)).thenReturn(existingUser);
+        when(userMapper.entityToResponse(existingUser)).thenReturn(updatedUserResponse);
+
+        UserResponse result = userService.updateUser(authentication, updateRequest);
 
- */
+        assertEquals(updatedUserResponse, result);
+        verify(authService, times(1)).encodePassword(newPassword);
+        verify(userRepository, times(1)).save(existingUser);
```

Comment:
```suggestion
        verify(authService).encodePassword(newPassword);
        verify(userRepository).save(existingUser);
```

![image](https://github.com/user-attachments/assets/77748819-7184-42f4-ab79-4f4014dbfb0f)